runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -530,6 +540,20 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 						arrayletElementOffset = 0;
 					}
 					break;
+
+				default :
+					// array size other than 1/2/4/8 (mainly for flattened types)
+					{
+						U_8* srcCursor = ((U_8*)srcLeafAddress) + arrayletElementOffset * elementSize;
+						U_8* destCursor = (U_8*)outerDestCursor;
+						outerCount -= innerCount;	// Decrement the outer count before looping
+						while(innerCount--) {
```

Comment:
we need double check. I believe this is not correct. `innerCount` counts in elements but you need in bytes, should be `innerCount * elementSize`